python/pyarrow/feather.py

```diff
@@ -165,25 +97,68 @@ def read_pandas(self, columns=None, use_threads=True):
             use_threads=use_threads)
 
 
-def write_feather(df, dest):
+def check_chunked_overflow(name, col):
+    if col.num_chunks == 1:
+        return
+
+    if col.type in (ext.binary(), ext.string()):
+        raise ValueError("Column '{}' exceeds 2GB maximum capacity of "
+                         "a Feather binary column. This restriction may be "
+                         "lifted in the future".format(name))
+    else:
+        # TODO(wesm): Not sure when else this might be reached
+        raise ValueError("Column '{}' of type {} was chunked on conversion "
+                         "to Arrow and cannot be currently written to "
+                         "Feather format".format(name, str(col.type)))
+
+
+def write_feather(df, dest, compression=None, compression_level=None,
+                  chunksize=None, version=2):
     """
     Write a pandas.DataFrame to Feather format.
 
     Parameters
     ----------
-    df : pandas.DataFrame
-        Dataframe to write out as feather format.
+    df : pandas.DataFrame or pyarrow.Table
+        Data to write out as Feather format.
     dest : str
         Local destination path.
+    compression : string, default None
+        {"zstd", "lz4", None}
+    compression_level : int, default None
+        Use a compression level particular to the chosen compressor. If None
+        use the default compression level
+    chunksize : int, default None
+        For V2 files, the size of chunks to split the data into. None means use
+        the default, which is currently 64K
+    version : int, default 2
+        Feather file version. Version 2 is the current. Version 1 is the more
+        limited legacy format
```

Comment:
@nealrichardson these are the new options that need to be exposed in R. It shouldn't be necessary to wrap any new C++ classes, only adding options to `write_feather`